File: src/src-frontend/src/types/document.ts
export interface DocumentEntry {
    id: string;
    filename: string;
    path: string;
    folder: '01' | '02' | '03' | '04' | '05';
    document_type: DocumentType;
    date: string | null;
    from: string | null;
    to: string | null;
    description: string;
    tags: string[];
    extracted_text: string | null;
    added_at: string;
}

export type DocumentType =
    | 'receipt'
    | 'letter'
    | 'email'
    | 'photo'
    | 'contract'
    | 'warranty'
    | 'court_form'
    | 'order'
    | 'other';

export const FOLDER_NAMES: Record<string, string> = {
    '01': 'Correspondence',
    '02': 'Evidence',
    '03': 'Legal',
    '04': 'Court',
    '05': 'Bundle',
};

export const DOCUMENT_TYPES: { value: DocumentType; label: string }[] = [
    { value: 'receipt', label: 'Receipt' },
    { value: 'letter', label: 'Letter' },
    { value: 'email', label: 'Email' },
    { value: 'photo', label: 'Photo' },
    { value: 'contract', label: 'Contract' },
    { value: 'warranty', label: 'Warranty' },
    { value: 'court_form', label: 'Court Form' },
    { value: 'order', label: 'Court Order' },
    { value: 'other', label: 'Other' },
];

File: src/src-tauri/src/models/document.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DocumentEntry {
    pub id: String,
    pub filename: String,
    pub path: String,
    pub folder: String,
    pub document_type: DocumentType,
    pub date: Option<String>,
    pub from: Option<String>,
    pub to: Option<String>,
    pub description: String,
    pub tags: Vec<String>,
    pub extracted_text: Option<String>,
    pub added_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum DocumentType {
    #[serde(rename = "receipt")]
    Receipt,
    #[serde(rename = "letter")]
    Letter,
    #[serde(rename = "email")]
    Email,
    #[serde(rename = "photo")]
    Photo,
    #[serde(rename = "contract")]
    Contract,
    #[serde(rename = "warranty")]
    Warranty,
    #[serde(rename = "court_form")]
    CourtForm,
    #[serde(rename = "order")]
    Order,
    #[serde(rename = "other")]
    Other,
}

impl Default for DocumentEntry {
    fn default() -> Self {
        Self {
            id: uuid::Uuid::new_v4().to_string(),
            filename: String::new(),
            path: String::new(),
            folder: "02".to_string(),
            document_type: DocumentType::Other,
            date: None,
            from: None,
            to: None,
            description: String::new(),
            tags: Vec::new(),
            extracted_text: None,
            added_at: chrono::Utc::now().to_rfc3339(),
        }
    }
}
